api-specifications.md

```diff
@@ -96,6 +104,162 @@ Create a course (anyone logged in). Used for outside Exchange.
 * Login required
 * Course already exists
 
+### /api/instructor: Course instructor management
+
+#### POST /api/instructor/&lt;course_id&gt;/&lt;instructor_id&gt;
+*Add an instructor to the course. (instructors only)*
+
+##### Request (HTTP POST data)
+```
+first_name=/*instructor name*/&
+last_name=/*instructor last name*/&
+email=/*instructor email*/
+```
+
+##### Response
+```javascript
+{
+    "success": true
+}
+```
+
+##### Error messages
+* Login required
+* Permission denied
+* Course not found
+
+#### GET /api/instructor/&lt;course_id&gt;/&lt;instructor_id&gt;
+*Gets information about a course instructor. (instructors+students)*
+
+##### Response
+```javascript
+{
+    "success": true,
+    "first_name": /*instructor name*/,
+    "last_name": /*instructor last name*/,
+    "email": /*instructor email*/
+}
+```
+
+##### Error messages
+* Login required
+* Permission denied
+* Course not found
+* Instructor not found
+* Removing self
+
+#### DELETE /api/instructor/&lt;course_id&gt;/&lt;instructor_id&gt;
+*Remove a course instructor (instructors only)*
+
+##### Error messages
+* Login required
+* Permission denied
+* Course not found
+* Instructor not found
+
+### /api/instructors: List course instructors
+
+#### GET /api/instructors/&lt;course_id&gt;
+*Gets information about all course instructors. (instructors+students)*
```

Comment:
If you have time: make sure to use "Third person singular" everywhere or nowhere. For example, on line 492 it is "**download** feedback on a", but here is it "**Gets** information ...". Make the first one "downloads" or this one "get". (also check all other API entry points)
Personally I prefer not using third person singular.